src/vm/zapsig.cpp

```diff
@@ -1127,7 +1127,11 @@ BOOL ZapSig::EncodeMethod(
     TypeHandle ownerType;
 
 #ifdef FEATURE_READYTORUN_COMPILER
-    if (IsReadyToRunCompilation())
+
+    // For methods encoded outside of the version bubble, we use pResolvedToken which describes the metadata token from which the method originated
+    // For tokens inside the version bubble we are not constrained by the contents of pResolvedToken and as such we skip this codepath
+    // FUTURE: This condition should likely be changed or reevaluated once support for smaller version bubbles is implemented.
+    if (IsReadyToRunCompilation() && !IsLargeVersionBubbleEnabled())
```

Comment:
>IsLargeVersionBubbleEnabled [](start = 38, length = 27)

I think this still isn't right. It should be something like IsReadyToRunCompilation() && (!IsLargeVersionBubbleEnabled() ||| !pMethod-.GetModule()-.IsInCurrentVersionBubble())

Reply:
Makes sense - we avoid using the metadata token in the case where large version bubbles are enabled, but the method being encoded is in the same assembly. Changed!